src/app/services/variable.service.ts

```diff
@@ -373,7 +372,17 @@ export class VariableService {
     this._selectedVariablesTreeUpdated = value;
   }
 
-  get selectedVariablesNumber(): number {
-    return this.selectedVariablesTree.filter(node => this.treeNodeService.isTreeNodeConcept(node)).length;
+  private updateNumberOfSelectedVariables(): number {
+    let num = 0;
+    this.variables.forEach((variable: Concept) => {
+      if (variable.selected) {
+        num++;
+      }
+    });
+    return num;
```

Comment:
this could be also simplified, something like this? 
`this.variables.filter((variable: Concept) => {variable.selected)).length`